parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMemoryManager.java

```diff
@@ -44,89 +44,146 @@
+
+    long rowGroupSize = expectedPoolSize / 2;
+    conf.setLong(ParquetOutputFormat.BLOCK_SIZE, rowGroupSize);
+
+    // the memory manager is not initialized until a writer is created
+    createWriter(1).close(null);
   }
 
-  @After
-  public void tearDown() throws Exception{
-    FileUtils.deleteDirectory(new File("target/test"));
+  @Test
+  public void testMemoryManagerUpperLimit() {
+    // Verify the memory pool size
+    // this value tends to change a little between setup and tests, so this
+    // validates that it is within 5% of the expected value
+    long poolSize = ParquetOutputFormat.getMemoryManager().getTotalMemoryPool();
+    Assert.assertTrue("Pool size should be within 5% of the expected value",
+        Math.abs(expectedPoolSize - poolSize) < (long) (expectedPoolSize * 0.05));
   }
 
   @Test
   public void testMemoryManager() throws Exception {
-    //Verify the adjusted rowGroupSize of writers
+    long poolSize = ParquetOutputFormat.getMemoryManager().getTotalMemoryPool();
+    long rowGroupSize = poolSize / 2;
+    conf.setLong(ParquetOutputFormat.BLOCK_SIZE, rowGroupSize);
+
+    Assert.assertTrue("Pool should hold 2 full row groups",
+        (2 * rowGroupSize) <= poolSize);
+    Assert.assertTrue("Pool should not hold 3 full row groups",
+        poolSize < (3 * rowGroupSize));
+
+    Assert.assertEquals("Allocations should start out at 0",
+        0, getTotalAllocation());
+
     RecordWriter writer1 = createWriter(1);
-    verifyRowGroupSize(rowGroupSize);
+    Assert.assertTrue("Allocations should never exceed pool size",
+        getTotalAllocation() <= poolSize);
+    Assert.assertEquals("First writer should be limited by row group size",
+        rowGroupSize, getTotalAllocation());
 
     RecordWriter writer2 = createWriter(2);
-    verifyRowGroupSize(rowGroupSize);
+    Assert.assertTrue("Allocations should never exceed pool size",
+        getTotalAllocation() <= poolSize);
+    Assert.assertEquals("Second writer should be limited by row group size",
+        2 * rowGroupSize, getTotalAllocation());
 
     RecordWriter writer3 = createWriter(3);
-    verifyRowGroupSize((int) Math.floor(expectPoolSize / 3));
+    Assert.assertTrue("Allocations should never exceed pool size",
+        getTotalAllocation() <= poolSize);
 
     writer1.close(null);
-    verifyRowGroupSize(rowGroupSize);
+    Assert.assertTrue("Allocations should never exceed pool size",
+        getTotalAllocation() <= poolSize);
+    Assert.assertEquals("Allocations should be increased to the row group size",
+        2 * rowGroupSize, getTotalAllocation());
 
     writer2.close(null);
-    verifyRowGroupSize(rowGroupSize);
+    Assert.assertTrue("Allocations should never exceed pool size",
+        getTotalAllocation() <= poolSize);
+    Assert.assertEquals("Allocations should be increased to the row group size",
+        rowGroupSize, getTotalAllocation());
 
     writer3.close(null);
+  }
```

Comment:
Should we add the same code that validates that getTotalAllocation() is 0 after closing writer3 to check that there's no leaks?